instrumentation/src/com/intellij/rt/coverage/instrumentation/kotlin/KotlinUtils.java

```diff
@@ -28,4 +40,33 @@ public static boolean isKotlinClass(Instrumenter context) {
     context.addProperty(KOTLIN_CLASS_LABEL, isKotlin);
     return isKotlin;
   }
+
+  private static boolean myKotlinEnabled = false;
```

Comment:
why can't you init the static field (normally named with "our" prefix) with  !"true".equals(System.getProperty("coverage.kotlin.disable", "false"));